Rules/CompatibilityRules/CompatibilityRule.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation.Language;
+using System.Text.RegularExpressions;
+using Microsoft.PowerShell.CrossCompatibility.Query;
+using Microsoft.PowerShell.CrossCompatibility.Utility;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    public abstract class CompatibilityRule : ConfigurableRule
```

Comment:
Created this to not repeat myself too much -- it just takes care of the profile loading and the patterns I used to keep the rule and visitor logic well contained

Reply:
The rules seem to have been executing for me. Is it something that occurs when rules are specified without `-IncludeRule` instead?

Reply:
Hmm, could be. I ran `Invoke-Scriptanalyzer -ScriptDefinition foo -Verbose` and saw that it loaded your module but not executed the rule.
This is good in the sense of overhead (i.e. the rule should not get run if it is not configured) but it would be ideal if it also did nor load your crosscompat module in this case

Reply:
I've now added the `[Export]` attribute for MEF, so the former should happen. I've also taken out the reference to CrossCompatibility in Engine.csproj. I'll see if it's loading the PowerShell module anywhere -- it shouldn't, since my intent was for them to be run quite separately (generation of profiles vs using them for lookup)